### ISSUE TITLE
nested destinations

### ISSUE THREAD

**Issue by sellmair** [sellmair/kompass#31]
Support destinations like 

```
class OuterDestination {
     @Destination
     class InnerDestination
}
```

**Comment by heinrichreimer** [comment]
Possibly related:
I want to use the following structure for storing destinations, as suggested in the readme:
```kotlin
sealed class AppDestination {
    @Destination
    class SampleDestination: AppDestination()
}
```
But it seems, the annotation processor fails to find the constructor and instead generates:
```kotlin
fun Kompass.Companion.bundleAsSampleDestination(bundle: Bundle): AppDestination.SampleDestination = SampleDestination()
```
which calls `SampleDestination()` instead of `AppDestination.SampleDestination()`

**Comment by sellmair** [comment]
Closed with prototype 0.2.0 #49
